src/treeview_commands.mli

```diff
@@ -1,3 +1,2 @@
-(** Register the ocaml-commands tree view and returns the provider as a
-    disposable. *)
-val register : Vscode.ExtensionContext.t -> Vscode.Disposable.t Promise.t
+(** Register the ocaml-commands tree view. *)
+val register : Vscode.ExtensionContext.t -> Extension_instance.t -> unit
```

Comment:
Why did you add the `Extension_instance.t` argument? I don't see it being used.

Reply:
Consistency of the API. All of the other `register` functions in the code take a pair of extension context and extension instance. We don't use it though, so I can remove it.

Reply:
It's better to get rid of it unless it adds some serious convenience. Much easier to reason about functions when we know what parameters they need.

Reply:
Done